modules/libcom/src/misc/epicsString.c

```diff
@@ -259,37 +259,42 @@ size_t epicsStrnLen(const char *s, size_t maxlen)
     return i;
 }
 
-int epicsStrGlobMatch(const char *str, const char *pattern)
+int epicsStrnGlobMatch(const char *str, size_t len, const char *pattern)
 {
-    const char *cp = NULL, *mp = NULL;
+    const char *mp = NULL;
+    size_t cp = 0, i = 0;
 
-    while ((*str) && (*pattern != '*')) {
-        if ((*pattern != *str) && (*pattern != '?'))
+    while ((i < len) && (str[i]) && (*pattern != '*')) {
+        if ((*pattern != str[i]) && (*pattern != '?'))
             return 0;
         pattern++;
-        str++;
+        i++;
     }
-    while (*str) {
+    while ((i < len) && str[i]) {
         if (*pattern == '*') {
             if (!*++pattern)
                 return 1;
             mp = pattern;
-            cp = str+1;
+            cp = i+1;
         }
-        else if ((*pattern == *str) || (*pattern == '?')) {
+        else if ((*pattern == str[i]) || (*pattern == '?')) {
             pattern++;
-            str++;
+            i++;
         }
         else {
             pattern = mp;
-            str = cp++;
+            i = cp++;
         }
     }
     while (*pattern == '*')
         pattern++;
     return !*pattern;
 }
 
+int epicsStrGlobMatch(const char *str, const char *pattern) {
+    return epicsStrnGlobMatch(str, (size_t)-1, pattern);
```

Comment:
The `(size_t)-1` here gives off a bad smell.

C99 requires stdint.h to define `SIZE_MAX` which IMHO would be the best solution here, but we never used stdint.h in EPICS because VxWorks didn't support C99 until VxWorks 6. I believe you are the last EPICS developer who has been trying to build the EPICS 7 core on VxWorks 5.5 – are you still doing that, or have you given it up?

Reply:
I am still using vxWorks 5.5.
I could define SIZE_MAX if it is not available otherwise. Or make a stdint.h header file for vxWorks 5.

Reply:
Hm... not even VxWorks 6.7 finds stdint.h. It works with 6.9.
VxWorks 6.7 indeed has stdint.h, twice in fact:
- diab/5.7.0.0/include/cnew/stdint.h
- vxworks-6.7/target/usr/h/stdint.h
But I use the gnu compiler, not the diab and target/usr/h is not in the include path.
I had to modify configure/os/CONFIG.Common.vxWorksCommon

Reply:
MSVC has been the other holdout for C99.  I guess vxworks is unlikely to have the POSIX alternative `inttypes.h`.  Personally I still use `(size_t)-1` in C code.  I also see an incorrect definition in `epicsSingletonMutex.cpp`, although present usage is inconsequential.
https://github.com/epics-base/epics-base/blob/b00130493fe695fad623e509bda9d507fccec019/modules/libcom/src/cxxTemplates/epicsSingletonMutex.cpp#L22-L24

Reply:
Maybe epicsTypes.h should `#include <stdint.h>` except on VxWorks <= 6.7 and before MSVC started, where we would define and document the things that we rely on, initially just `SIZE_MAX`? Eventually I can see us redefining some the epics types in terms of the stdint.h types, once we insist on C99 compiler support (EPICS 7.1).

Reply:
I found this today"
../../src/factory/FieldCreateFactory.cpp:577: warning: converting of negative value `-0x00000000000000001' to `size_t'
So it is not unusual to cast -1 to size_t. At least I do it explicitly :-)

Reply:
Turns out that VxWoks 6.9 automatically includes stdint.h via vxWorks.h which is included using a compiler option in all vxWorks builds. Thus checking if SIZE_MAX is defined for vxWorks is sufficient and no complicated version check is required.